classes/03class/exercises/c03-aws03/alanlima/README.md

```diff
@@ -0,0 +1,135 @@
+                    "TTL": 600,
+                    "ResourceRecords": [
+                        {
+                            "Value": "devopsacademy-lb-1691285263.ap-southeast-2.elb.amazonaws.com"
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+    ```
+    
+    * Using UPSERT because it is practical and I may need to update it anyway
+
+- Checking that it actually works..
+
+    ```bash
+    $ ssh ec2-user@13.236.232.76
+    Last login: Thu Jul  9 08:57:30 2020 from 122-148-135-233.sta.wbroadband.net.au
+    
+           __|  __|_  )
+           _|  (     /   Amazon Linux 2 AMI
+          ___|\___|___|
+    
+    https://aws.amazon.com/amazon-linux-2/
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-104.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-104.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    ```
+
+- Post any findings, how option you decided to go with, records created, domain created, and anything else that you find interesting.
+
+    - I was going to do the public hosted zone, but it took a while until I receive the e-mail for the free domain registration. Then, I started the private, make it works, got excited, received a verification e-mail, did the public zone as well 😅
+    - AWS will set the hosted zone to private when VPC is specified, even if you set PrivateZone=False. I needed to recreate the route a couple of times until I release that. The CLI didn't give any warning. 😢
+    - I found the docs to update the recordset not as clear as the other docs. I could not find exactly how I should create the A record to link to the ALB. I ended up creating it on the console and listed the resources to check how it was created.
+    - I would expect some small delay until the route 53 propagate the changes, but as soon as I set it, I was able to connect to my private hosted zone. Impressive 👌👌.
+    - Overall, I found it quite easy to set up a private hosted zone. When I started, I thought that I would need to execute a thousand commands.
+    - Forgot to set the key pair to my instances so I needed to recreate the launch configuration, ASG and update the LB register again. 🙄 *I am getting good in recreating things 🤣🤣*
+    - I cannot set CNAME record with AliasTarget; the connection will be refused. But it works as expected for the standard way with TTL and string value to DNS alias.
```

Comment:
Good test.

CNAME translates a name to another name. So basically you are telling R53 to map your name to the ALB DNS name. 

** The problem with this approach is that AWS does not guarantee that the DNS name of the ALB won't change, so better to use `alias`. **

Reply:
thanks, I will take note on that.